---
title: "PRS prediction EUROPEAN"
author: "Ruth Topless"
date: "`r format(Sys.Date())`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F)
library(DT)
library(tidyverse)
library(magrittr)
numcols = c("OR","LCI","UCI", "P","N")
library(ggplot2)
library(ggrepel)

results= function(test, i=2){
  matrixoutput = matrix(nrow=1, ncol=6)
  matrixoutput[1] <- names(coefficients(test)[i])
  matrixoutput[2] <- round(as.numeric(exp(coefficients(test))[i]),3)#OR
  matrixoutput[3] <- round(as.numeric(exp(coef(summary(test))[i, 1] - (1.96 * coef(summary(test))[i, 2]))),3) #LCI
  matrixoutput[4] <- round(as.numeric(exp(coef(summary(test))[i, 1] + (1.96 * coef(summary(test))[i, 2]))),3) #UCI
  matrixoutput[5] <- as.numeric(coef(summary(test))[i,c(4)]) #p value
  matrixoutput[6] <- test$df.null+1
  return ((matrixoutput))
}

# set colour palette
#devtools::install_github("G-Thomson/Manu")
library(Manu)

gwas_palette <- c("#7ACCD7", "#115896", "#7C6C65", "#4C4C53", "#BA2F00", "#B865A1", "#6AA086")
gwas_palette <- sort(gwas_palette)

```

Questions:
Which Gout affection status =  the same as went into GWAS.

adjusted by age at visit 0
289 SNPs
  Missing SNPS = "rs3750508"   "rs565410312"
  no proxies used


Extra notes: I cut the PRS range into 10 equal sized portions. I left in the lower outlier as it helps the upper limits make sense (if I cut it out the boundaries change and the prevalence of gout drops in the top group and all the graphs look a bit shit). all the prevalence/freq graphs represent the full 10 cut. In the association data I added the bottom groups together so that there were some gout cases in that group so we could get some stats. the same with the top group in women. 


```{r genotypes, eval=FALSE}
genotypes<-read.delim("/Volumes/biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/UKBB_GWAS_2022_all.txt")
allsnps<-read.delim("/Volumes/archive/merrimanlab/major_gwas_paper_archive/results_for_paper/loci_list/full_indepSNP_summary_1jun2022.txt")
allsnps[allsnps==""]=NA

EUR<-allsnps[which(allsnps$EUR=="EUR"),]
EUR$risk_allele<-NA
EUR$risk_allele[which(EUR$OR_EUR>=1)]<-EUR$MinorAllele[which(EUR$OR_EUR>=1)]
EUR$risk_allele[which(EUR$OR_EUR<1)]<-EUR$MajorAllele[which(EUR$OR_EUR<1)]
EUR$oth_allele<-NA
EUR$oth_allele[which(EUR$OR_EUR>=1)]<-EUR$MajorAllele[which(EUR$OR_EUR>=1)]
EUR$oth_allele[which(EUR$OR_EUR<1)]<-EUR$MinorAllele[which(EUR$OR_EUR<1)]
EUR$effect_used<-NA
EUR$effect_used[which(EUR$OR_EUR>=1)]<-EUR$OR_EUR[which(EUR$OR_EUR>=1)]
EUR$effect_used[which(EUR$OR_EUR<1)]<-1/EUR$OR_EUR[which(EUR$OR_EUR<1)]

alllist<-c("id",EUR$SNP)
genotypes<-genotypes[,which(colnames(genotypes) %in% alllist)]
genotypes[genotypes==""]=NA
genotypes[genotypes=="00"]=NA

alllist[!alllist %in% colnames(genotypes)]

snplist<-colnames(genotypes)[2:ncol(genotypes)]
for(snp in snplist){
  RISK_geno <- paste0(EUR[which(EUR$SNP == snp), "risk_allele"], EUR[which(EUR$SNP == snp), "risk_allele"], collapse = "")
  ALT_geno <-  paste0(EUR[which(EUR$SNP == snp), "oth_allele"], EUR[which(EUR$SNP == snp), "oth_allele"], collapse = "")
  HET1_geno <- paste0(EUR[which(EUR$SNP == snp), "risk_allele"], EUR[which(EUR$SNP == snp), "oth_allele"], collapse = "")
  HET2_geno <- paste0(EUR[which(EUR$SNP == snp), "oth_allele"], EUR[which(EUR$SNP == snp), "risk_allele"], collapse = "")
  genotypes[which(genotypes[, snp] == RISK_geno),snp]=2
  genotypes[which(genotypes[, snp] == ALT_geno),snp]=0
  genotypes[which(genotypes[, snp] == HET1_geno),snp]=1
  genotypes[which(genotypes[, snp] == HET2_geno),snp]=1
}

genotypes[,snplist] %<>% lapply(function(x) as.numeric(as.character(x)))  
for(snp in snplist){
  genotypes[,snp]<-genotypes[,snp]*EUR$effect_used[which(EUR$SNP==snp)]
}

genotypes$sum<-rowSums(genotypes[2:ncol(genotypes)], na.rm = T)
genotypes$nacount<-apply(genotypes[2:(ncol(genotypes)-1)], 1, function(x) sum(is.na(x)))

genotypes$GRS<-(genotypes$sum/(289-genotypes$nacount))*(289)
write.table(genotypes[,c("id","GRS")], "/Volumes/biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/UKBB_EURO_grs_289_for_pheWAS.txt", row.names = F, quote = F, na="", sep="\t")

```


```{r phenotypes}
#GRS pheWAS in ukbiobank
pheno<-read.delim("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/UKBioBank/Phenotypes/data/inital_pheno_visit0_decoded.txt", stringsAsFactors = F)
pheno2<-pheno %>% select(f_eid, f_31_0_0, f_34_0_0,f_21000_0_0, f_189_0_0, f_54_0_0, f_21001_0_0, f_20116_0_0, f_21003_0_0)
colnames(pheno2)<-c("f_eid", "SEX", "YEAR","ETHNICITY", "TOWNSEND", "SITE", "BMI", "SMOKER", "AGE")
affstat<-read.delim("/Volumes/archive/merrimanlab/raid_backup/UKbiobank/genetic_files/gout_gwas_covar.20210201.covar")
pheno2<-merge(pheno2,affstat[,c("IID","plink_goutaff")], by.x="f_eid",by.y="IID")
pheno2$gout_AFFSTAT=NA
pheno2$gout_AFFSTAT[which(pheno2$plink_goutaff==2)]=1
pheno2$gout_AFFSTAT[which(pheno2$plink_goutaff==1)]=0
pheno2<-pheno2[is.na(pheno2$gout_AFFSTAT)==F,]

pheno2$ETHNICCLASS<-pheno2$ETHNICITY
pheno2$ETHNICCLASS <- plyr::mapvalues(pheno2$ETHNICCLASS, from = c("African","White and Black African", "Black or Black British","Caribbean", "Any other Black background","White and Black Caribbean", "Asian or Asian British","Chinese","Indian","Pakistani","White and Asian","Any other Asian background", "Bangladeshi", "British","Any other white background", "Irish","White", "Prefer not to answer","Any other mixed background","Do not know","Mixed","Other ethnic group"), to = c("BLACK","BLACK","BLACK","BLACK","BLACK","BLACK","ASIAN","ASIAN","ASIAN","ASIAN","ASIAN","ASIAN","ASIAN","WHITE", "WHITE","WHITE","WHITE", "OTHER","OTHER","OTHER","OTHER","OTHER"))
pheno2<-pheno2[which(pheno2$ETHNICCLASS=="WHITE"),]

goutprs<-read.delim("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/UKBB_EURO_grs_289_for_pheWAS.txt")
colnames(goutprs)<-c("f_eid","GoutPRS")
pheno2<-merge(pheno2,goutprs, by.x="f_eid", all.x = T)
pheno2<-pheno2[is.na(pheno2$GoutPRS)==F,]

from = range(pheno2$GoutPRS, na.rm = T)[1] 
to = range(pheno2$GoutPRS, na.rm = T)[2]
by = (to-from)/10
breaksgout <- seq( from = from, to = to, by = by )
# bucketing values into bins

pheno2<-pheno2 %>% mutate(GoutPRS_bins = cut(pheno2$GoutPRS, breaks=breaksgout, include.lowest=TRUE, right=FALSE))

breaksgout2<-breaksgout[c(1,4:11)]
pheno2<-pheno2 %>% mutate(GoutPRS_bins2 = cut(pheno2$GoutPRS, breaks=breaksgout2, include.lowest=TRUE, right=FALSE))

breaksgout3<-breaksgout[c(1,4:9,11)]
pheno2<-pheno2 %>% mutate(GoutPRS_bins3 = cut(pheno2$GoutPRS, breaks=breaksgout3, include.lowest=TRUE, right=FALSE))
```

## Part A Gout PRS bins

```{r}
sumfilegout<-as.data.frame(table(pheno2$GoutPRS_bins))
sumfilegout$controls<-table(pheno2$GoutPRS_bins, pheno2$gout_AFFSTAT)[1:10]
sumfilegout$gout<-table(pheno2$GoutPRS_bins, pheno2$gout_AFFSTAT)[11:20]
sumfilegout$prevalencegout<-round(x = (sumfilegout$gout/sumfilegout$Freq)*100, digits = 3)
sumfilegout %>%  datatable(., caption = "GoutPRS bins and gout prevalence", escape = F, options = list(pageLength = 30, autoWidth = TRUE) , rownames = F)

f<-ggplot(sumfilegout, aes(Var1,Freq))+geom_col(fill=gwas_palette[1])+theme_light()+
  xlab("Gout risk score bin")+ylab("Frequency")+
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20)))
f

ggsave("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/GoutBins_freq.svg")
a<-sumfilegout %>% mutate(Var1_number = as.numeric(Var1)) %>% ggplot( aes(Var1_number,prevalencegout))+geom_line()+theme_light()+
  xlab("Gout risk score bin")+ylab("Gout prevalence (%)")+
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20))) + scale_x_continuous(labels = as.character(sumfilegout$Var1), breaks = 1:nlevels(sumfilegout$Var1))
a

ggsave("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/GoutBins_prevalence.svg")
```


## Part B Gout PRS bins
```{r}
resulttable <- as.data.frame(matrix(ncol=6, nrow=0))
colnames(resulttable)<-c("name","OR","LCI","UCI", "P","N" )

pheno2$GoutPRS_bins2<-factor(pheno2$GoutPRS_bins2, levels =c("[290,301)","[237,269)", "[269,280)", "[280,290)", "[301,312)", "[312,322)", "[322,333)", "[333,344]"))
test<-glm(gout_AFFSTAT~GoutPRS_bins2+AGE+SEX, data=pheno2, family = 'binomial') 

for(b in seq(2:11)){
    resulttable[b,]<-results(test=test,i=b)
}
#replace intercept info with reference bin data
resulttable[1,]<-c("[290,301)","1","1","1",NA, NA)
#insert nice names
resulttable[,1]<-c("[290,301)","[237,269)", "[269,280)", "[280,290)", "[301,312)", "[312,322)", "[322,333)", "[333,344]","Age","Sex")

resulttable[,numcols] %<>% lapply(function(x) as.numeric(as.character(x)))  
#add significance  stars for plotting 
resulttable$sig<-NA
resulttable$sig[which(resulttable$P<0.05)]="*"
resulttable$sig[which(resulttable$P<0.01)]="**"
resulttable$sig[which(resulttable$P<0.001)]="***"

#reorder results
resulttable$name <- factor(resulttable$name, levels = c("[237,269)", "[269,280)", "[280,290)", "[290,301)", "[301,312)", "[312,322)", "[322,333)", "[333,344]","Age","Sex"))
resulttable$labelpos<-ifelse(resulttable$OR <= 1, resulttable$LCI, resulttable$UCI)


ggplot(data=resulttable, aes(x=name, y=OR)) + theme_light()+
  geom_bar(stat="identity", color="black", fill= "#115896") +
  labs(title = "European risk score", y= "Odds Ratio (Log scale)", x = "Risk Factor") +
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20))) + 
  scale_y_log10(limits = c(0.09,45)) + 
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.2, position=position_dodge(.9)) +
  geom_text_repel(aes(x=name, y=labelpos, label = sig), na.rm = T, nudge_y = ifelse(resulttable$labelpos <=1, -0.09, +0.01))


library(svglite)
ggsave("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/GoutBins_barchart.svg")

resulttable %>%  datatable(., caption = "Logisitic regression Gout ~ GRS bins", escape = F, options = list(pageLength = 30, autoWidth = TRUE) , rownames = F)


```



# Males

```{r,eval=T}
males<-pheno2[which(pheno2$SEX==1),]
sumfilegout_males<-as.data.frame(table(males$GoutPRS_bins))
sumfilegout_males$controls<-table(males$GoutPRS_bins, males$gout_AFFSTAT)[1:10]
sumfilegout_males$gout<-table(males$GoutPRS_bins, males$gout_AFFSTAT)[11:20]
sumfilegout_males$prevalencegout<- round(x = (sumfilegout_males$gout/sumfilegout_males$Freq)*100, digits = 3)

f<-ggplot(sumfilegout_males, aes(Var1,Freq))+geom_col(fill=gwas_palette[2])+theme_light()+
  xlab("Gout risk score bin")+ylab("Gout prevalence (%)")+
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20)))
f
ggsave("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/EUR_full_GoutBins_freq_males.svg")

a<-sumfilegout_males %>% mutate(Var1_number = as.numeric(Var1)) %>% ggplot( aes(Var1_number,prevalencegout))+geom_line()+theme_light()+
  xlab("Gout risk score bin")+ylab("Gout prevalence (%)")+
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20))) + scale_x_continuous(labels = as.character(sumfilegout$Var1), breaks = 1:nlevels(sumfilegout$Var1))
a
ggsave("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/EUR_full_GoutBins_prevalence_males.svg")

sumfilegout_males %>%  datatable(., caption = "GoutPRS bins and gout prevalence in males", escape = F, options = list(pageLength = 30, autoWidth = TRUE) , rownames = F)

resulttable_males <- as.data.frame(matrix(ncol=6, nrow=0))
colnames(resulttable_males)<-c("name","OR","LCI","UCI", "P","N" )
males$GoutPRS_bins2<-factor(males$GoutPRS_bins2, levels =c("[290,301)","[237,269)", "[269,280)", "[280,290)", "[301,312)", "[312,322)", "[322,333)", "[333,344]"))
test<-glm(males$gout_AFFSTAT ~ males$GoutPRS_bins2 + males$AGE, data=males, family = 'binomial') 

for(b in seq(1:9)){
    resulttable_males[b,]<-results(test=test,i=b)
}

#replace intercept info with reference bin data
resulttable_males[1,]<-c("[290,301)","1","1","1",NA,NA,NA)
#insert nice names
resulttable_males[,1]<-c("[290,301)","[237,269)", "[269,280)", "[280,290)", "[301,312)", "[312,322)", "[322,333)", "[333,344]", "Age")

resulttable_males[,numcols] %<>% lapply(function(x) as.numeric(as.character(x)))  

resulttable_males$sig<-NA
resulttable_males$sig[which(resulttable_males$P<0.05)]="*"
resulttable_males$sig[which(resulttable_males$P<0.01)]="**"
resulttable_males$sig[which(resulttable_males$P<0.001)]="***"

resulttable_males$name <- factor(resulttable_males$name, levels  = c("[237,269)", "[269,280)","[280,290)", "[290,301)", "[301,312)", "[312,322)", "[322,333)", "[333,344]", "Age"))
resulttable_males$labelpos<-ifelse(resulttable_males$OR < 1, resulttable_males$LCI, resulttable_males$UCI)

ggplot(data=resulttable_males, aes(x=name, y=OR))  + theme_light()+
  geom_bar(stat="identity", colour="black", fill= gwas_palette[2]) +
  labs(title = "European risk score in Males", y= "Odds Ratio (Log scale)", x = "Risk Factor") +
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20))) + 
  scale_y_log10(limits = c(0.09,45)) + 
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.2, position=position_dodge(.9)) +
  geom_text_repel(aes(x=name, y=labelpos, label = sig), na.rm = T, nudge_y = ifelse(resulttable_males$labelpos <=1, -0.09, +0.01))


ggsave("/Volumes/biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/MaleCohort_barchart.svg")

resulttable_males %>%  datatable(., caption = "Logisitic regression Gout ~ GRS bins in males ", escape = F, options = list(pageLength = 30, autoWidth = TRUE) , rownames = F)
```

# Females

```{r,eval=T}
females<-pheno2[which(pheno2$SEX==0),]

sumfilegout_females<-as.data.frame(table(females$GoutPRS_bins))
sumfilegout_females$controls<-table(females$GoutPRS_bins, females$gout_AFFSTAT)[1:10]
sumfilegout_females$gout<-table(females$GoutPRS_bins, females$gout_AFFSTAT)[11:20]
sumfilegout_females$prevalencegout<- round(x = (sumfilegout_females$gout/sumfilegout_females$Freq)*100, digits = 3)

f<-ggplot(sumfilegout_females, aes(Var1,Freq))+geom_col(fill=gwas_palette[3])+theme_light()+
  xlab("Gout risk score bin")+ylab("Frequency")+
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20)))
f
ggsave("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/EUR_full_GoutBins_freq_females.svg")
a<-sumfilegout_females %>% mutate(Var1_number = as.numeric(Var1)) %>% ggplot( aes(Var1_number,prevalencegout))+geom_line()+theme_light()+
  xlab("Gout risk score bin")+ylab("Gout prevalence (%)")+
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20))) +    scale_x_continuous(labels = as.character(sumfilegout$Var1), breaks = 1:nlevels(sumfilegout$Var1))
a
ggsave("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/EUR_full_GoutBins_prevalence_females.svg")

sumfilegout_females %>%  datatable(., caption = "GoutPRS bins and gout prevalence in females", escape = F, options = list(pageLength = 30, autoWidth = TRUE) , rownames = F)

resulttable_females <- as.data.frame(matrix(ncol=6, nrow=0))
colnames(resulttable_females)<-c("name","OR","LCI","UCI", "P","N" )
females$GoutPRS_bins3<-factor(females$GoutPRS_bins3, levels =c("[290,301)","[237,269)", "[269,280)", "[280,290)", "[301,312)", "[312,322)", "[322,344]"))
test<-glm(females$gout_AFFSTAT ~ females$GoutPRS_bins3 + females$AGE, data=females, family = 'binomial') 

for(b in seq(1:8)){
    resulttable_females[b,]<-results(test=test,i=b)
}

#replace intercept info with reference bin data
resulttable_females[1,]<-c("[290,301)","1","1","1",NA,NA)
#insert nice names
resulttable_females[,1]<-c("[290,301)","[237,269)", "[269,280)","[280,290)",  "[301,312)", "[312,322)", "[322,344]", "Age")

resulttable_females[,numcols] %<>% lapply(function(x) as.numeric(as.character(x)))  

resulttable_females$sig<-NA
resulttable_females$sig[which(resulttable_females$P<0.05)]="*"
resulttable_females$sig[which(resulttable_females$P<0.01)]="**"
resulttable_females$sig[which(resulttable_females$P<0.001)]="***"

resulttable_females$name <- factor(resulttable_females$name, levels  = c("[237,269)", "[269,280)","[280,290)", "[290,301)", "[301,312)", "[312,322)", "[322,344]", "Age"))
resulttable_females$labelpos<-ifelse(resulttable_females$OR < 1, resulttable_females$LCI, resulttable_females$UCI)


ggplot(data=resulttable_females, aes(x=name, y=OR)) + theme_light()+
  geom_bar(stat="identity", color="black", fill= gwas_palette[3]) +
  labs(title = "European risk score", y= "Odds Ratio (Log scale)", x = "Risk Factor") +
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20))) + 
  scale_y_log10(limits = c(0.09,45)) + 
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.2, position=position_dodge(.9)) +
  geom_text_repel(aes(x=name, y=labelpos, label = sig), na.rm = T, nudge_y = ifelse(resulttable_females$labelpos <=1, -0.09, +0.01))


ggsave("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/FemaleCohort_barchart.svg")

resulttable_females %>%  datatable(., caption = "Logisitic regression Gout ~ GRS bins in females ", escape = F, options = list(pageLength = 30, autoWidth = TRUE) , rownames = F)
```
# Male GWAS PRS
```{r, eval=F}
genotypes<-read.delim("/Volumes/biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/UKBB_GWAS_2022_all.txt")
allsnps<-read.delim("/Volumes/archive/merrimanlab/major_gwas_paper_archive/results_for_paper/loci_list/male_indepSNP_summary_1jun2022.txt")
allsnps[allsnps==""]=NA

EUR<-allsnps[which(allsnps$EUR=="EUR"),]
EUR$risk_allele<-NA
EUR$risk_allele[which(EUR$OR_EUR>=1)]<-EUR$MinorAllele[which(EUR$OR_EUR>=1)]
EUR$risk_allele[which(EUR$OR_EUR<1)]<-EUR$MajorAllele[which(EUR$OR_EUR<1)]
EUR$oth_allele<-NA
EUR$oth_allele[which(EUR$OR_EUR>=1)]<-EUR$MajorAllele[which(EUR$OR_EUR>=1)]
EUR$oth_allele[which(EUR$OR_EUR<1)]<-EUR$MinorAllele[which(EUR$OR_EUR<1)]
EUR$effect_used<-NA
EUR$effect_used[which(EUR$OR_EUR>=1)]<-EUR$OR_EUR[which(EUR$OR_EUR>=1)]
EUR$effect_used[which(EUR$OR_EUR<1)]<-1/EUR$OR_EUR[which(EUR$OR_EUR<1)]

alllist<-c("id",EUR$SNP)
genotypes<-genotypes[,which(colnames(genotypes) %in% alllist)]
genotypes[genotypes==""]=NA
genotypes[genotypes=="00"]=NA

alllist[!alllist %in% colnames(genotypes)]

snplist<-colnames(genotypes)[2:ncol(genotypes)]
for(snp in snplist){
  RISK_geno <- paste0(EUR[which(EUR$SNP == snp), "risk_allele"], EUR[which(EUR$SNP == snp), "risk_allele"], collapse = "")
  ALT_geno <-  paste0(EUR[which(EUR$SNP == snp), "oth_allele"], EUR[which(EUR$SNP == snp), "oth_allele"], collapse = "")
  HET1_geno <- paste0(EUR[which(EUR$SNP == snp), "risk_allele"], EUR[which(EUR$SNP == snp), "oth_allele"], collapse = "")
  HET2_geno <- paste0(EUR[which(EUR$SNP == snp), "oth_allele"], EUR[which(EUR$SNP == snp), "risk_allele"], collapse = "")
  genotypes[which(genotypes[, snp] == RISK_geno),snp]=2
  genotypes[which(genotypes[, snp] == ALT_geno),snp]=0
  genotypes[which(genotypes[, snp] == HET1_geno),snp]=1
  genotypes[which(genotypes[, snp] == HET2_geno),snp]=1
}


genotypes[,snplist] %<>% lapply(function(x) as.numeric(as.character(x)))  
for(snp in snplist){
  genotypes[,snp]<-genotypes[,snp]*EUR$effect_used[which(EUR$SNP==snp)]
}

genotypes$sum<-rowSums(genotypes[2:ncol(genotypes)], na.rm = T)
genotypes$nacount<-apply(genotypes[2:(ncol(genotypes)-1)], 1, function(x) sum(is.na(x)))

genotypes$GRS<-(genotypes$sum/(246-genotypes$nacount))*(246)
write.table(genotypes[,c("id","GRS")], "/Volumes/biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/UKBB_EURO_MALE_grs_246_for_pheWAS.txt", row.names = F, quote = F, na="", sep="\t")

```



```{r,eval=T}
goutprs<-read.delim("/Volumes/biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/UKBB_EURO_MALE_grs_246_for_pheWAS.txt")
colnames(goutprs)<-c("f_eid","GoutPRS")
pheno2<-pheno2[,c("f_eid","SEX","YEAR","ETHNICITY","TOWNSEND","SITE","BMI","SMOKER","AGE","plink_goutaff","gout_AFFSTAT","ETHNICCLASS")]

pheno2<-merge(pheno2,goutprs, by.x="f_eid", all.x = T)
pheno2<-pheno2[is.na(pheno2$GoutPRS)==F,]
males<-pheno2[which(pheno2$SEX==1),]


from = range(males$GoutPRS, na.rm = T)[1] 
to = range(males$GoutPRS, na.rm = T)[2]
by = (to-from)/10
breaksgout <- seq( from = from, to = to, by = by )
# bucketing values into bins

males<-males %>% mutate(GoutPRS_bins = cut(males$GoutPRS, breaks=breaksgout, include.lowest=TRUE, right=FALSE))

breaksgout2<-breaksgout[c(1,3:11)]
males<-males %>% mutate(GoutPRS_bins2 = cut(males$GoutPRS, breaks=breaksgout2, include.lowest=TRUE, right=FALSE))

breaksgout3<-breaksgout[c(1,4:9,11)]
males<-males %>% mutate(GoutPRS_bins3 = cut(males$GoutPRS, breaks=breaksgout3, include.lowest=TRUE, right=FALSE))


sumfilegout_males<-as.data.frame(table(males$GoutPRS_bins))
sumfilegout_males$controls<-table(males$GoutPRS_bins, males$gout_AFFSTAT)[1:10]
sumfilegout_males$gout<-table(males$GoutPRS_bins, males$gout_AFFSTAT)[11:20]
sumfilegout_males$prevalencegout<- round(x = (sumfilegout_males$gout/sumfilegout_males$Freq)*100, digits = 3)

f<-ggplot(sumfilegout_males, aes(Var1,Freq))+geom_col(fill=gwas_palette[2])+theme_light()+
  xlab("Gout risk score bin")+ylab("Gout prevalence (%)")+
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20)))
f
ggsave("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/EUR_full_GoutBins_freq_males.svg")

a<-sumfilegout_males %>% mutate(Var1_number = as.numeric(Var1)) %>% ggplot( aes(Var1_number,prevalencegout))+geom_line()+theme_light()+
  xlab("Gout risk score bin")+ylab("Gout prevalence (%)")+
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20))) + scale_x_continuous(labels = as.character(sumfilegout$Var1), breaks = 1:nlevels(sumfilegout$Var1))
a
ggsave("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/EUR_full_GoutBins_prevalence_males.svg")

sumfilegout_males %>%  datatable(., caption = "GoutPRS bins and gout prevalence in males", escape = F, options = list(pageLength = 30, autoWidth = TRUE) , rownames = F)

resulttable_males <- as.data.frame(matrix(ncol=6, nrow=0))
colnames(resulttable_males)<-c("name","OR","LCI","UCI", "P","N" )
males$GoutPRS_bins2<-factor(males$GoutPRS_bins2, levels =c("[237,246)","[199,218)", "[218,227)", "[227,237)",  "[246,255)", "[255,265)", "[265,274)", "[274,283)", "[283,293]"))
test<-glm(males$gout_AFFSTAT ~ males$GoutPRS_bins2 + males$AGE, data=males, family = 'binomial') 

for(b in seq(1:10)){
    resulttable_males[b,]<-results(test=test,i=b)
}

#replace intercept info with reference bin data
resulttable_males[1,]<-c("[237,246)","1","1","1",NA,NA)
#insert nice names
resulttable_males[,1]<-c("[237,246)","[199,218)", "[218,227)", "[227,237)",  "[246,255)", "[255,265)", "[265,274)", "[274,283)", "[283,293]", "Age")

resulttable_males[,numcols] %<>% lapply(function(x) as.numeric(as.character(x)))  

resulttable_males$sig<-NA
resulttable_males$sig[which(resulttable_males$P<0.05)]="*"
resulttable_males$sig[which(resulttable_males$P<0.01)]="**"
resulttable_males$sig[which(resulttable_males$P<0.001)]="***"

resulttable_males$name <- factor(resulttable_males$name, levels  = c("[199,218)", "[218,227)", "[227,237)","[237,246)", "[246,255)", "[255,265)", "[265,274)", "[274,283)", "[283,293]", "Age"))
resulttable_males$labelpos<-ifelse(resulttable_males$OR < 1, resulttable_males$LCI, resulttable_males$UCI)

ggplot(data=resulttable_males, aes(x=name, y=OR))  + theme_light()+
  geom_bar(stat="identity", colour="black", fill= gwas_palette[2]) +
  labs(title = "European risk score in Males", y= "Odds Ratio (Log scale)", x = "Risk Factor") +
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20))) + 
  scale_y_log10(limits = c(0.005,100)) + 
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.2, position=position_dodge(.9)) +
  geom_text_repel(aes(x=name, y=labelpos, label = sig), na.rm = T, nudge_y = ifelse(resulttable_males$labelpos <=1, -0.09, +0.01))


ggsave("/Volumes/biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/MaleCohort_barchart.svg")

resulttable_males %>%  datatable(., caption = "Logisitic regression Gout ~ GRS bins in males ", escape = F, options = list(pageLength = 30, autoWidth = TRUE) , rownames = F)
```
# Female GWAS PRS

```{r, eval=F}
genotypes<-read.delim("/Volumes/biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/UKBB_GWAS_2022_all.txt")
allsnps<-read.delim("/Volumes/archive/merrimanlab/major_gwas_paper_archive/results_for_paper/loci_list/female_indepSNP_summary_1jun2022.txt")
allsnps[allsnps==""]=NA

EUR<-allsnps[which(allsnps$EUR=="EUR"),]
EUR$risk_allele<-NA
EUR$risk_allele[which(EUR$OR_EUR>=1)]<-EUR$MinorAllele[which(EUR$OR_EUR>=1)]
EUR$risk_allele[which(EUR$OR_EUR<1)]<-EUR$MajorAllele[which(EUR$OR_EUR<1)]
EUR$oth_allele<-NA
EUR$oth_allele[which(EUR$OR_EUR>=1)]<-EUR$MajorAllele[which(EUR$OR_EUR>=1)]
EUR$oth_allele[which(EUR$OR_EUR<1)]<-EUR$MinorAllele[which(EUR$OR_EUR<1)]
EUR$effect_used<-NA
EUR$effect_used[which(EUR$OR_EUR>=1)]<-EUR$OR_EUR[which(EUR$OR_EUR>=1)]
EUR$effect_used[which(EUR$OR_EUR<1)]<-1/EUR$OR_EUR[which(EUR$OR_EUR<1)]

alllist<-c("id",EUR$SNP)
genotypes<-genotypes[,which(colnames(genotypes) %in% alllist)]
genotypes[genotypes==""]=NA
genotypes[genotypes=="00"]=NA

alllist[!alllist %in% colnames(genotypes)]

snplist<-colnames(genotypes)[2:ncol(genotypes)]
for(snp in snplist){
  RISK_geno <- paste0(EUR[which(EUR$SNP == snp), "risk_allele"], EUR[which(EUR$SNP == snp), "risk_allele"], collapse = "")
  ALT_geno <-  paste0(EUR[which(EUR$SNP == snp), "oth_allele"], EUR[which(EUR$SNP == snp), "oth_allele"], collapse = "")
  HET1_geno <- paste0(EUR[which(EUR$SNP == snp), "risk_allele"], EUR[which(EUR$SNP == snp), "oth_allele"], collapse = "")
  HET2_geno <- paste0(EUR[which(EUR$SNP == snp), "oth_allele"], EUR[which(EUR$SNP == snp), "risk_allele"], collapse = "")
  genotypes[which(genotypes[, snp] == RISK_geno),snp]=2
  genotypes[which(genotypes[, snp] == ALT_geno),snp]=0
  genotypes[which(genotypes[, snp] == HET1_geno),snp]=1
  genotypes[which(genotypes[, snp] == HET2_geno),snp]=1
}


genotypes[,snplist] %<>% lapply(function(x) as.numeric(as.character(x)))  
for(snp in snplist){
  genotypes[,snp]<-genotypes[,snp]*EUR$effect_used[which(EUR$SNP==snp)]
}

genotypes$sum<-rowSums(genotypes[2:ncol(genotypes)], na.rm = T)
genotypes$nacount<-apply(genotypes[2:(ncol(genotypes)-1)], 1, function(x) sum(is.na(x)))

genotypes$GRS<-(genotypes$sum/(14-genotypes$nacount))*(14)
write.table(genotypes[,c("id","GRS")], "/Volumes/biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/UKBB_EURO_FEMALE_grs_14_for_pheWAS.txt", row.names = F, quote = F, na="", sep="\t")

```

```{r,eval=T}
goutprs<-read.delim("/Volumes/biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/UKBB_EURO_FEMALE_grs_14_for_pheWAS.txt")
colnames(goutprs)<-c("f_eid","GoutPRS")
pheno2<-pheno2[,c("f_eid","SEX","YEAR","ETHNICITY","TOWNSEND","SITE","BMI","SMOKER","AGE","plink_goutaff","gout_AFFSTAT","ETHNICCLASS")]

pheno2<-merge(pheno2,goutprs, by.x="f_eid", all.x = T)
pheno2<-pheno2[is.na(pheno2$GoutPRS)==F,]
females<-pheno2[which(pheno2$SEX==1),]

from = range(females$GoutPRS, na.rm = T)[1] 
to = range(females$GoutPRS, na.rm = T)[2]
by = (to-from)/10
breaksgout <- seq( from = from, to = to, by = by )
# bucketing values into bins

females<-females %>% mutate(GoutPRS_bins = cut(females$GoutPRS, breaks=breaksgout, include.lowest=TRUE, right=FALSE))

breaksgout2<-breaksgout[c(1,3:11)]
females<-females %>% mutate(GoutPRS_bins2 = cut(females$GoutPRS, breaks=breaksgout2, include.lowest=TRUE, right=FALSE))

breaksgout3<-breaksgout[c(1,4:9,11)]
females<-females %>% mutate(GoutPRS_bins3 = cut(females$GoutPRS, breaks=breaksgout3, include.lowest=TRUE, right=FALSE))


sumfilegout_females<-as.data.frame(table(females$GoutPRS_bins))
sumfilegout_females$controls<-table(females$GoutPRS_bins, females$gout_AFFSTAT)[1:10]
sumfilegout_females$gout<-table(females$GoutPRS_bins, females$gout_AFFSTAT)[11:20]
sumfilegout_females$prevalencegout<- round(x = (sumfilegout_females$gout/sumfilegout_females$Freq)*100, digits = 3)

f<-ggplot(sumfilegout_females, aes(Var1,Freq))+geom_col(fill=gwas_palette[3])+theme_light()+
  xlab("Gout risk score bin")+ylab("Frequency")+
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20)))
f
ggsave("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/EUR_full_GoutBins_freq_females.svg")
a<-sumfilegout_females %>% mutate(Var1_number = as.numeric(Var1)) %>% ggplot( aes(Var1_number,prevalencegout))+geom_line()+theme_light()+
  xlab("Gout risk score bin")+ylab("Gout prevalence (%)")+
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20))) +    scale_x_continuous(labels = as.character(sumfilegout$Var1), breaks = 1:nlevels(sumfilegout$Var1))
a
ggsave("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/EUR_full_GoutBins_prevalence_females.svg")

sumfilegout_females %>%  datatable(., caption = "GoutPRS bins and gout prevalence in females", escape = F, options = list(pageLength = 30, autoWidth = TRUE) , rownames = F)

resulttable_females <- as.data.frame(matrix(ncol=6, nrow=0))
colnames(resulttable_females)<-c("name","OR","LCI","UCI", "P","N" )
females$GoutPRS_bins2<-factor(females$GoutPRS_bins2, levels =c("[14.5,16.9)","[4.98,9.76)", "[9.76,12.1)","[12.1,14.5)",  "[16.9,19.3)", "[19.3,21.7)", "[21.7,24.1)", "[24.1,26.5)", "[26.5,28.9]"))
test<-glm(females$gout_AFFSTAT ~ females$GoutPRS_bins2 + females$AGE, data=females, family = 'binomial') 

for(b in seq(1:10)){
    resulttable_females[b,]<-results(test=test,i=b)
}

#replace intercept info with reference bin data
resulttable_females[1,]<-c("[14.5,16.9)","1","1","1",NA,NA)
#insert nice names
resulttable_females[,1]<-c("[14.5,16.9)","[4.98,9.76)", "[9.76,12.1)","[12.1,14.5)",  "[16.9,19.3)", "[19.3,21.7)", "[21.7,24.1)", "[24.1,26.5)", "[26.5,28.9]", "Age")

resulttable_females[,numcols] %<>% lapply(function(x) as.numeric(as.character(x)))  

resulttable_females$sig<-NA
resulttable_females$sig[which(resulttable_females$P<0.05)]="*"
resulttable_females$sig[which(resulttable_females$P<0.01)]="**"
resulttable_females$sig[which(resulttable_females$P<0.001)]="***"

resulttable_females$name <- factor(resulttable_females$name, levels  = c("[4.98,9.76)", "[9.76,12.1)","[12.1,14.5)", "[14.5,16.9)", "[16.9,19.3)", "[19.3,21.7)", "[21.7,24.1)", "[24.1,26.5)", "[26.5,28.9]", "Age"))
resulttable_females$labelpos<-ifelse(resulttable_females$OR < 1, resulttable_females$LCI, resulttable_females$UCI)


ggplot(data=resulttable_females, aes(x=name, y=OR)) + theme_light()+
  geom_bar(stat="identity", color="black", fill= gwas_palette[3]) +
  labs(title = "European risk score", y= "Odds Ratio (Log scale)", x = "Risk Factor") +
  theme(plot.title = element_text(size=14, face="bold", hjust = 0.5), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none", axis.title.x = element_text(margin = margin(t = 20))) + 
  scale_y_log10(limits = c(0.09,30)) + 
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.2, position=position_dodge(.9)) +
  geom_text_repel(aes(x=name, y=labelpos, label = sig), na.rm = T, nudge_y = ifelse(resulttable_females$labelpos <=1, -0.09, +0.01))


ggsave("/Volumes//biochemistry/Lab_Groups/merrimanlab/Merriman_Documents/Ruth/2021_projects/post_GWAS_analysis/PRS/FemaleCohort_barchart.svg")

resulttable_females %>%  datatable(., caption = "Logisitic regression Gout ~ GRS bins in females ", escape = F, options = list(pageLength = 30, autoWidth = TRUE) , rownames = F)
```
